plugin/inlay_hint.py

```diff
@@ -21,7 +21,8 @@ class LspToggleInlayHintsCommand(LspWindowCommand):
 
     def __init__(self, window: sublime.Window) -> None:
         super().__init__(window)
-        window.settings().set('lsp_show_inlay_hints', userprefs().show_inlay_hints)
+        settings = userprefs()
+        window.settings().setdefault('lsp_show_inlay_hints', False if settings is None else settings.show_inlay_hints)
```

Comment:
So this still seems to be setting `False` in the case when settings are `None` (and no previous value is stored). Which can different from actual value from userprefs.

Reply:
Yes, but it only happens if the command is initialized before `plugin_loaded` and if the window doesn't already have a value stored. I think it's not safe to use API methods before plugin_loaded unless they are explicitly documented to be available on import time. Though iirc there might have been some changes to that in a past ST update (?)

Reply:
I found it in the changelog that from ST 4171 all functions are available at import time (also tested and confirmed).

So now it uses the API method to read from the settings file if the build is new enough: https://github.com/sublimelsp/LSP/pull/2571/commits/7f3da69bffa93d3d38afcf746b6610dd68f86efc